### DISCUSSION TITLE
View Transitions API support

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#11669] [Ideas]
Since the new [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API) is starting to be adopted by frontend frameworks like [Nuxt](https://nuxt.com/docs/getting-started/transitions#view-transitions-api-experimental), it would be really nice to have a Tailwind way of adding the CSS classes and properties.

Yea, I like living on the cutting edge 😛 

Right now, I end up having to use arbitrary values and custom CSS:

```html
<template>
    <NuxtImg
        class="rounded [view-transition-name:selected-film]"
    />

    <h1 class="text-lg [view-transition-name:header]">
        {{ movie.title }}
    </h1>
</template>

<style>
    ::view-transition-old(header),
    ::view-transition-new(header) {
        width: auto;
    }
</style>
```

**Comment by alexanderson1993** [comment]
View transitions are now supported by Astro and React Router/Remix. As this API gets more usage and becomes more popular, it would be excellent to be able to enable and configure the transitions without leaving my markup.

**Comment by RyannGalea** [comment]
Definitely should be on the roadmap.

**Comment by frederichoule** [comment]
[View Transitions API is also supported by SvelteKit](https://svelte.dev/blog/view-transitions), and the Svelte community heavily relies on Tailwind. I think it should be added to the roadmap!

**Comment by RafaelZasas** [comment]
Is there any update on this?

**Comment by mhornbacher** [comment]
There is a nice little plugin if your setup supports those at https://github.com/ekafyi/tailwindcss-view-transitions

**Comment by FifthWit** [comment]
I absolutely support this

**Comment by ansarizafar** [comment]
Any update?

**Comment by BrahmVanH** [comment]
I've found myself hoping for this as well. Exploring view transitions in sveltekit has me wishing I there was a TW class for the view-transition-name property for inline use.

**Comment by hckhanh** [comment]
+[

**Comment by MrLoh** [comment]
Can you please stop +1 spamming this issue and use the upvote functionality.

**Reply by isaacfennell** [reply]
@MrLoh +1
